### PR TITLE
Bump to Mutiny 2.6.1 and Mutiny Vert.x bindings 3.13.1

### PR DESCRIPTION
**Pull request by jponge**
Mutiny 2.6.1 depends on JCTools 4.0.5 and now uses unpadded+atomic queues that simplify native compilation. We can thus remove substitutions to support the native compilation of Mutiny.

### CODE REVIEW

**Review by gsmet — APPROVED** [review]

**Review by zakkak — COMMENTED** [review]
:tada: Thanks @jponge ! 

The changes look good to me but the CI doesn't seem happy.

Any idea why the amqp tests are failing in JVM-mode?

**Comment by cescoffier** [comment]
Yeah, it looks odd that the AMQP tests start failing...

**Comment by jponge** [comment]
I'll have a look locally at those tests

**Comment by jponge** [comment]
Ok now we have a proper issue 🤣 

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.615 s <<< FAILURE! -- in io.quarkus.smallrye.reactivemessaging.amqp.SecuredAmqpTest
[ERROR] io.quarkus.smallrye.reactivemessaging.amqp.SecuredAmqpTest -- Time elapsed: 1.615 s <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:708)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:285)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:661)
	... 1 more
Caused by: java.lang.NoClassDefFoundError: org/jctools/queues/atomic/unpadded/SpscUnboundedAtomicUnpaddedArrayQueue
	at io.smallrye.mutiny.helpers.queues.Queues.createSpscUnboundedArrayQueue(Queues.java:25)
	at io.smallrye.mutiny.helpers.queues.Queues.lambda$unbounded$5(Queues.java:77)
	at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.create(UnicastProcessor.java:59)
	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$TriggerSubscriber.<init>(MultiRetryWhenOp.java:178)
	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp.subscribe(MultiRetryWhenOp.java:44)
	at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp.subscribe(MultiRetryWhenOp.java:74)
	at io.smallrye.mutiny.groups.MultiSubscribe.withSubscriber(MultiSubscribe.java:70)
	at io.smallrye.mutiny.operators.multi.MultiOnFailureInvoke.subscribe(MultiOnFailureInvoke.java:29)
	at io.smallrye.mutiny.groups.MultiSubscribe.withSubscriber(MultiSubscribe.java:70)
	at io.smallrye.mutiny.operators.multi.MultiMapOp.subscribe(MultiMapOp.java:24)
	at io.smallrye.mutiny.groups.MultiSubscribe.withSubscriber(MultiSubscribe.java:70)
	at io.smallrye.reactive.messaging.providers.locals.ContextOperator$ContextMulti.subscribe(ContextOperator.java:34)
	at io.smallrye.mutiny.groups.MultiSubscribe.withSubscriber(MultiSubscribe.java:70)
	at io.smallrye.mutiny.operators.multi.MultiMapOp.subscribe(MultiMapOp.java:24)
	at io.smallrye.mutiny.operators.AbstractMulti.subscribe(AbstractMulti.java:60)
	at io.smallrye.mutiny.operators.multi.MultiConcatMapOp.subscribe(MultiConcatMapOp.java:54)
	at io.smallrye.mutiny.groups.MultiSubscribe.withSubscriber(MultiSubscribe.java:70)
	at io.smallrye.mutiny.operators.multi.MultiOnFailureInvoke.subscribe(MultiOnFailureInvoke.java:29)
	at io.smallrye.mutiny.operators.AbstractMulti.subscribe(AbstractMulti.java:60)
	at io.smallrye.reactive.messaging.providers.SubscriberMediator.run(SubscriberMediator.java:149)
	at io.smallrye.reactive.messaging.providers.wiring.Wiring$SubscriberMediatorComponent.materialize(Wiring.java:732)
	at io.smallrye.reactive.messaging.providers.wiring.Graph.lambda$materialize$10(Graph.java:100)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at io.smallrye.reactive.messaging.providers.wiring.Graph.materialize(Graph.java:99)
	at io.smallrye.reactive.messaging.providers.extension.MediatorManager.start(MediatorManager.java:263)
	at io.smallrye.reactive.messaging.providers.extension.MediatorManager_ClientProxy.start(Unknown Source)
	at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:53)
	at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_qTrMuLFyQ1IvGfeSxRVitl6CCBQ.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:80)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:156)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:107)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 6 more
Caused by: java.lang.ClassNotFoundException: org.jctools.queues.atomic.unpadded.SpscUnboundedAtomicUnpaddedArrayQueue
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:526)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:476)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:526)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:476)
	... 41 more
```

**Comment by jponge** [comment]
Looks like for some reason the wrong JCTools version is used, let me dig it

**Comment by jponge** [comment]
Ok we need to align Reactive Messaging to the latest Mutiny and JCTools, otherwise Maven does not _always_ resolve to JCTools 4.0.5 but 4.0.2.

Let's put this PR on hold until a new release is ready 😃  

/cc @ozangunalp

**Comment by gsmet** [comment]
@jponge JCTools should be in the BOM.

**Comment by jponge** [comment]
Ok let me offer both a PR to Reactive Messaging + add JCTools to the application BOM then

**Comment by ozangunalp** [comment]
@jponge We don't have direct dependency on jctools in reactive messaging. We only pull it through Mutiny.

**Comment by jponge** [comment]
Updated, let's see what CI thinks

File: bom/application/pom.xml [hunk 1]
```diff
@@ -138,7 +138,8 @@
         <brotli4j.version>1.16.0</brotli4j.version>
         <reactive-streams.version>1.0.4</reactive-streams.version>
         <jboss-logging.version>3.6.0.Final</jboss-logging.version>
-        <mutiny.version>2.6.0</mutiny.version>
+        <mutiny.version>2.6.1</mutiny.version>
+        <jctools-core.version>4.0.5</jctools-core.version>
```
Comment (zakkak) [inline bom/application/pom.xml:142]: In the PR description you mention version 4.0.4. Is this on purpose or a typo?

Comment (jponge) [reply bom/application/pom.xml:142]: Good catch, that was a typo!

**Review by cescoffier — APPROVED** [review]

**Pull request merged**